packages/core/src/sync-realtime/service.ts

```diff
@@ -441,6 +448,11 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
     );
     newBlocks.push(newBlock);
 
+    // Detect re-org
+    if (newBlocks[0].parentHash !== latestLocalBlock.hash) {
```

Comment:
Do we also need to check that the missing range doesn't have a forked block in it? Eg if `newBlocks` has length > 1?